actfw_core/task/join.py

```diff
@@ -0,0 +1,74 @@
+import traceback
+from queue import Empty, Full, Queue
+from threading import Thread
+
+from .task import Task
+
+
+class Join(Task):
+
+    """Join Task."""
+
+    def __init__(self):
+        """"""
+        super(Join, self).__init__()
+        self.running = True
+        self.in_queues = []
+        self.out_queues = []
+        self.out_queue_id = 0
+
+    def _is_running(self):
+        return self.running
+
+    def _add_in_queue(self, q):
+        self.in_queues.append(q)
+
+    def _add_out_queue(self, q):
+        self.out_queues.append(q)
+
+    def _inlet(self):
+        while self._is_running():
+            try:
+                yield tuple(in_queue.get(timeout=1) for in_queue in self.in_queues)
```

Comment:
Consider the case `in_queue = [a, b, c]`.

If `b` is empty, this code consumes `a`'s frames. This makes the "freshness" of the contents of `a` and `c` different.
Note that some producers, e.g. `actfw_raspberrypi.PiCameraCapture`, [maintain frames even after pushing a frame to the successors](https://github.com/Idein/actfw-raspberrypi/blob/release-1.4.0/actfw_raspberrypi/capture.py#L35-L41), but this behavior depends on the implementations and I think we should not rely on it. (IIUC, even `actfw_raspberrypi.PiCameraCapture` case, it maintain by "best effort". If an instance has `outqueues = [x, y]` and `x` consumes fast and `y` slow, it does not update some frames for `y`.)

Is this behavior what you intended?

Reply:
https://docs.python.org/ja/3/library/queue.html#queue.Queue.get
FYI, `Queue.get(block=True, timeout=None)` remedy the behavior.

Reply:
No, my intention is "Waiting for all value without blocking". So the current implementation is wrong, but `block=True` is not a suitable solution. I'll fix it.

Reply:
https://github.com/Idein/actfw-core/pull/15/commits/61d00468d9d6b6e7c11a6f53902f633a2d080518